### PR TITLE
Feature/us7315 say hi login confirmation

### PR DESCRIPTION
**Pull request by ilpadre**

### CODE REVIEW

File: src/app/components/pin-details/say-hi/say-hi.component.ts [hunk 1]
```diff
@@ -24,14 +26,54 @@ export class SayHiComponent {
...
+              this.router.navigate(['/member-said-hi']); // Change this to generic confirmation page component
+            },
+            err => {
+              // redirect to error page
+            }
+
+          );
+        },
+        err => {
+
```
Comment (JoeKer1) [inline src/app/components/pin-details/say-hi/say-hi.component.ts]: these blank errors need stuff in them?

File: src/app/components/authentication/authentication.component.html [hunk 2]
```diff
@@ -52,7 +52,7 @@
     <p class="font-size-smaller">Or <a routerLink="/register">register for an account</a>.</p>
 
     <footer class="page-footer">
-      <a routerLink="/" class="btn btn-gray pull-left">Back</a>
+      <a (click) = "back();" class="btn btn-gray pull-left">Back</a>
```
Comment (JoeKer1) [inline src/app/components/authentication/authentication.component.html]: Does that semicolon need to be there?

File: src/app/components/authentication/authentication.component.ts [hunk 3]
```diff
@@ -24,14 +24,14 @@ export class AuthenticationComponent implements OnInit {
...
-  private forgotPasswordUrl: string;
+  private forgotPasswordUrl: string; 
   private helpUrl: string;
 
   constructor(
     private api: APIService,
     private fb: FormBuilder,
     private router: Router,
-    private redirectService: LoginRedirectService,
+    public redirectService: LoginRedirectService,
```
Comment (JoeKer1) [inline src/app/components/authentication/authentication.component.ts]: I've only ever seen these as private, does it need to be public? Not that it matters in javascript :)

File: src/app/components/pin-details/gathering/gathering-requests/gathering-requests.component.spec.ts [hunk 4]
```diff
@@ -69,7 +69,7 @@ describe('GatheringRequestsComponent', () => {
         (<jasmine.Spy>mockGroupService.getGroupRequests).and.returnValue(Observable.of([ {inquiryId: 1}, {inquiryId: 2}, {inquiryId: 3, placed: true}]));
         (<jasmine.Spy>mockStateService.setLoading).and.returnValue(true);
       comp.ngOnInit();
-      expect(comp.inquiries.length).toBe(2);
+      expect(comp.getInquiries().length).toBe(2);
```
Comment (JoeKer1) [inline src/app/components/pin-details/gathering/gathering-requests/gathering-requests.component.spec.ts]: Jim gave Doug a different fix for this that didn't require a new method. Jasmine Jim

File: src/app/components/authentication/authentication.component.ts (hunk 3 above)
Comment (ilpadre) [inline src/app/components/authentication/authentication.component.ts]: I needed to make it public so I could use it in a test.

**Pull request merged**